apps/oc_chef_authz/src/oc_chef_policy.erl

```diff
@@ -0,0 +1,182 @@
+    list_policies_for_org.
+
+bulk_get_query() ->
+    %% TODO: do we need this?
+    ok.
+
+new_record(OrgId, AuthzId, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData),
+    Id = ej:get({<<"id">>}, PolicyData, chef_object_base:make_org_prefix_id(OrgId, Name)),
+    PolicyGroup = ej:get({<<"policy_group">>}, PolicyData),
+    #oc_chef_policy{
+       id = Id,
+       authz_id = AuthzId,
+                       org_id = OrgId,
+                       name = Name,
+                       policy_group = PolicyGroup,
+      serialized_object = PolicyData}.
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
+
+set_created(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+set_updated(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+is_indexed() ->
+    false.
+
+ejson_for_indexing(#oc_chef_policy{}, _EjsonTerm) ->
+   {[]}.
+
+update_from_ejson(#oc_chef_policy{} = Policy, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData, name(Policy)),
+    Group = ej:get({<<"policy_group">>}, PolicyData, policy_group(Policy)),
+    Policy#oc_chef_policy{name = Name,
+                         policy_group = Group}.
+
+fields_for_update(#oc_chef_policy{
+                     last_updated_by = LastUpdatedBy,
+                     serialized_object = SerializedObject
+                                 } = Policy) ->
+    [name(Policy), policy_group(Policy), LastUpdatedBy, SerializedObject].
+
+
+fields_for_fetch(#oc_chef_policy{org_id = OrgId} = Policy) ->
+    [name(Policy), policy_group(Policy), OrgId].
+
+record_fields() ->
+    record_info(fields, oc_chef_policy).
+
+list(#oc_chef_policy{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+update(#oc_chef_policy{
+                      org_id = _OrgId,
+                      authz_id = _PolicyAuthzId,
+                      last_updated_by = _AuthzId
+                     } = _Record, _CallbackFun) ->
+    1.
+
+
+parse_binary_json(Bin) ->
+    {ok, chef_json:decode_body(Bin)}.
+
+
+fetch(#oc_chef_policy{} = Record, CallbackFun) ->
+    chef_object:default_fetch(Record, CallbackFun).
+
+
+flatten(#oc_chef_policy{
+          id = Id,
+          authz_id = AuthzId,
+          org_id = OrgId,
+          name = Name,
+          policy_group = Group,
+          last_updated_by = LastUpdatedBy,
+          serialized_object = SerializedObject}) ->
+    Compressed = chef_db_compression:compress(oc_chef_policy, jiffy:encode(SerializedObject)),
+    [Id, AuthzId, OrgId, Name, Group, LastUpdatedBy, Compressed].
+
+
+assemble_policy_ejson(#oc_chef_policy{
+                         serialized_object = SerializedObject
+                        }, _OrgName) ->
+    chef_db_compression:decompress(SerializedObject).
```

Comment:
FYI, here's a bit from my in-progress RFC about policy groups and their purpose:

> The means by which a node is associated to a Policyfile lock is designed such that it is possible to have more than one revision of a Policyfile lock active in a Chef organization at one time. That is, it must be possible to update the
Policyfile lock for only some nodes with a given functional role. For example, a configuration code change may deploy a newer version of some important software that must be tested in a non-production environment before production deployment, or may cause a service restart that temporarily reduces capacity.

> It is also desirable that the Policyfile lock document itself not contain any information about which nodes it is applied to, as this simplifies the overall design and makes it easy to create tooling to manage promotion of a Policyfile lock through the phases of a deployment cycle.

> To accommodate these two constraints, Policyfile locks are associated to a node via both the policy name and policy group. The policy group models the phases of a deployment cycle, such as "development", "staging" and "production." Each policy group can have a different revision of a Policyfile lock, which enables configuration code to be updated in one policy group without any effect on the configuration code in other policy groups.